.github/workflows/binding-checks.yml

```diff
@@ -0,0 +1,70 @@
+name: Binding Checks
+
+on:
+  pull_request:
+    paths:
+      - # Any new or changed canisters
+        'dfx.json'
+      - # Any new or changed .did files
+        '**/*.did'
+      - # Scripts, GitHub actions that contain 'backend' in their path.
+        '**/*backend*'
+      - # The backend source code
+        'src/backend/**'
+      - 'src/shared/**'
+      - # Rust files such as Cargo.lock, Cargo.toml and rust-toolchain.toml
+        '**/Cargo*'
+      - '**/*rust*'
+      - # This workflow
+        '.github/workflows/binding-checks.yml'
+  workflow_dispatch:
+
+jobs:
+
+  generate:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Cache backend wasm
+        uses: actions/cache@v4
+        id: backend-wasm-cache
+        with:
+          path: |
+            target/wasm32-unknown-unknown/release/backend.wasm
+          key: ${{ runner.os }}-backend-wasm-${{ hashFiles('Cargo.lock', 'rust-toolchain.toml', 'src/backend/**/*', 'src/shared/**/*') }}
+      - name: Cache cargo dependencies
+        if: steps.backend-wasm-cache.outputs.cache-hit != 'true'
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-backend-tests-${{ hashFiles('Cargo.lock', 'rust-toolchain.toml') }}
+      - name: Prepare
+        uses: ./.github/actions/prepare
+      - name: Install dfx
+        uses: dfinity/setup-dfx@main
+      - name: Install binstall
+        run: |
+          BINSTALL_VERSION="1.8.0"
+          curl -L --proto '=https' --tlsv1.2 -sSf "https://github.com/cargo-bins/cargo-binstall/releases/download/v${BINSTALL_VERSION}/cargo-binstall-x86_64-unknown-linux-musl.tgz" | tar -xvzf - cargo-binstall
+          ./cargo-binstall -y --force "cargo-binstall@$BINSTALL_VERSION"
+          rm cargo-binstall
+      - name: Install candid-extractor
+        run: cargo binstall --no-confirm candid-extractor@0.1.4 && candid-extractor --version
+      - name: Generate bindings
+        run: npm run generate
+      - name: Check bindings
+        run: |
+          test -z "$(git status --porcelain)" || {
+                  echo "FIX: Please run npm run generate"
```

Comment:
An sub-sub-sub-sub-sub-sub-important idea, maybe "Please execute npm run generate" to avoid the content "run npm run"

Reply:
Good idea.  Done.